src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3345,13 +3392,15 @@ internal uint GetValEscape(BoundExpression expr, uint scopeOfTheContainingExpres
                     return Binder.CallingMethodScope;
 
                 case BoundKind.DiscardExpression:
-                    return ((BoundDiscardExpression)expr).ValEscape;
+                    return Binder.CallingMethodScope;
 
                 case BoundKind.DeconstructValuePlaceholder:
-                    return ((BoundDeconstructValuePlaceholder)expr).ValEscape;
+                case BoundKind.InterpolatedStringArgumentPlaceholder:
+                case BoundKind.AwaitableValuePlaceholder:
```

Comment:
nit: Not related to this PR, consider asserting the not bound nodes deriving from PlaceholderBase make it to the default case (they should all be handled explicitly)

Reply:
The `default` case includes `Debug.Assert(false);`.